libraries/cms/application/site.php

```diff
@@ -563,79 +563,49 @@ protected function initialiseApp($options = array())
 		$user = JFactory::getUser();
 
 		// If the user is a guest we populate it with the guest user group.
+		// SMZ: why do we need this???
 		if ($user->guest)
 		{
 			$guestUsergroup = JComponentHelper::getParams('com_users')->get('guest_usergroup', 1);
 			$user->groups = array($guestUsergroup);
 		}
 
-		// If a language was specified it has priority, otherwise use user or default language settings
-		if (empty($options['language']))
-		{
-			// Detect the specified language
-			$lang = $this->input->getString('language', null);
-
-			// Make sure that the user's language exists
-			if ($lang && JLanguage::exists($lang))
-			{
-				$options['language'] = $lang;
-			}
-			else
-			{
-				// Detect cookie language
-				$lang = $this->input->cookie->get(JApplicationHelper::getHash('language'), null, 'string');
+		// Determine the language settings.
+		// SMZ: $options['language'] is for sure empty at this time...
 
-				// Make sure that the user's language exists
-				if ($lang && JLanguage::exists($lang))
-				{
-					$options['language'] = $lang;
-				}
-				else
-				{
-					// Detect user language
-					$lang = $user->getParam('language');
-
-					// Make sure that the user's language exists
-					if ($lang && JLanguage::exists($lang))
-					{
-						$options['language'] = $lang;
-					}
-				}
-			}
-		}
+		// See if a language was specified: this has top priority.
+		// SMZ: But when/how this can be set?
+		$options['language'] = self::checkLang($this->input->getString('language', null));
```

Comment:
Via the URL. &language=de-DE is valid and has been possible in the past, even without the languagefilter plugin enabled.